### ISSUE TITLE
Better support for row level security filter with Distributed

### ISSUE THREAD

**Issue by KochetovNicolai** [ClickHouse/ClickHouse#28334]
Now, it is possible to use row level security filter with Distributed table. In order to apply it, you need to set up filter condition on every remote host (using xml config or RBAC). However, it may be not convenient to support a full set of users on every data node in coherence.

Possible solution:
* Allow to specify row level filter for distributed table
* Send filtering condition as a part of remote query (e.g. in client info)

This require a change in native protocol. If remote host is old, or row level filter cannot be applied, throw an exception.

**Comment by filimonov** [comment]
It was of the goals of https://github.com/ClickHouse/ClickHouse/pull/13156

**Comment by KochetovNicolai** [comment]
@filimonov as I understand, in #13156 we can use initial_user to run query on remote host. As I understand, you still need to have initial_user on remote. Or if you don't row level filter will be ignored?

**Comment by amosbird** [comment]
> Send filtering condition as a part of remote query (e.g. in client info)

Is it possible to just AND row level filter and utilize short circuit evaluation?

**Comment by KochetovNicolai** [comment]
> Is it possible to just AND row level filter and utilize short circuit evaluation?

Now we don't have an option to specify which functions must be short circuit. We could probably force short circut for all remote query. However I don't think it is a good solution e.g. because row level condition will be visible for user.

**Comment by vitlibar** [comment]
Yes, a normal user ideally should not be aware of the existence of row policies. And also it's important to pass row policies outside the query because otherwise there can be ways to trick row policies with just aliases.

**Comment by amosbird** [comment]
> there can be ways to trick row policies with just aliases.

Interesting...

**Comment by fuCtor** [comment]
I have same problem with distributed tables. In my case, I have some thin instance which provide only distributed tables for users and only this instance contains all data access restrictions, users and roles.
